broker/src/main/java/io/camunda/zeebe/broker/bootstrap/PartitionManagerStep.java

```diff
@@ -55,6 +55,9 @@ void startupInternal(
                                   () ->
                                       forwardExceptions(
                                           () -> {
+                                            final var adminApi =
+                                                brokerStartupContext.getAdminApiService();
+                                            adminApi.injectPartitionManager(partitionManager);
                                             final var adminService =
                                                 brokerStartupContext.getBrokerAdminService();
```

Comment:
:thought_balloon: Unrelated to this PR. Two AdminServices can be confusing :smile: At some point, we should move the old AdminService to the new AdminApiRequestHandler and expose those operations in the gateway similar to "stepDown".

Reply:
Yeah, totally agree. I was thinking about just reusing the `BrokerAdminService` but it currently only works over all partitions at once, which _would_ fit the current goal for the gateway api (just one `rebalanceLeaders` command, cluster wide) but isn't as flexible in the future.